flash.sh

```diff
@@ -0,0 +1,298 @@
+
+  if [ -z "${DRIVE}" ]; then
+    echo You must specify a SD card drive
+  fi
+
+  umount $DRIVE
+
+  if [ -e "out/target/product/$DEVICE/$2.img" ]; then
+		dd if=out/target/product/${DEVICE}/$2.img of=${DRIVE} bs=1M
+	fi
+}
+
+flash_gecko()
+{
+	delete_extra_gecko_files_on_device &&
+	run_adb push $GECKO_OBJDIR/dist/b2g /system/b2g &&
+	return 0
+}
+
+flash_gaia()
+{
+	GAIA_MAKE_FLAGS="ADB=\"$ADB\""
+	USER_VARIANTS="user(debug)?"
+	# We need to decide where to push the apps here.
+	# If the VARIANTS is user or userdebug, send them to /system/b2g.
+	# or, we will try to connect the phone and see where Gaia was installed
+	# and try not to push to the wrong place.
+	if [[ "$VARIANT" =~ $USER_VARIANTS ]]; then
+		# Gaia's build takes care of remounting /system for production builds
+		echo "Push to /system/b2g ..."
+		GAIA_MAKE_FLAGS+=" GAIA_INSTALL_PARENT=/system/b2g"
+	else
+		echo "Detect GAIA_INSTALL_PARENT ..."
+		# This part has been re-implemented in Gaia build script (bug 915484),
+		# XXX: Remove this once we no longer support old Gaia branches.
+		# Install to /system/b2g if webapps.json does not exist, or
+		# points any installed app to /system/b2g.
+		run_adb wait-for-device
+		if run_adb shell 'cat /data/local/webapps/webapps.json || echo \"basePath\": \"/system\"' | grep -qs '"basePath": "/system' ; then
+			echo "Push to /system/b2g ..."
+			GAIA_MAKE_FLAGS+=" GAIA_INSTALL_PARENT=/system/b2g"
+		else
+			echo "Push to /data/local ..."
+			GAIA_MAKE_FLAGS+=" GAIA_INSTALL_PARENT=/data/local"
+		fi
+	fi
+	make -C gaia push $GAIA_MAKE_FLAGS
+
+	# For older Gaia without |push| target,
+	# run the original |install-gaia| target.
+	# XXX: Remove this once we no longer support old Gaia branches.
+	if [[ $? -ne 0 ]]; then
+		make -C gaia install-gaia $GAIA_MAKE_FLAGS
+	fi
+	return $?
+}
+
+while [ $# -gt 0 ]; do
+	case "$1" in
+	"-s")
+		ADB_FLAGS+="-s $2"
+		shift
+		;;
+  "-f")
+		FULLFLASH=true
+		;;
+	"-h"|"--help")
+		echo "Usage: $0 [-s device] [-f] [project]"
+		exit 0
+		;;
+	"-"*)
+		echo "$0: Unrecognized option: $1"
+		exit 1
+		;;
+	*)
+		PROJECT=$1
+		;;
+	esac
+	shift
+done
+
+if $FULLFLASH; then
+  sd_partition &&
+  write_images
+  exit $?
+else
+  case "$PROJECT" in
+  "shallow")
+  	resp=`run_adb root` || exit $?
+  	[ "$resp" != "adbd is already running as root" ] && run_adb wait-for-device
+  	run_adb shell stop b2g &&
+  	run_adb remount &&
+  	flash_gecko &&
+  	flash_gaia &&
+  	update_time &&
+  	echo Restarting B2G &&
+  	run_adb shell start b2g
+  	exit $?
+  	;;
+
+  "gecko")
+  	resp=`run_adb root` || exit $?
+  	[ "$resp" != "adbd is already running as root" ] && run_adb wait-for-device
+  	run_adb shell stop b2g &&
+  	run_adb remount &&
+  	flash_gecko &&
+  	echo Restarting B2G &&
+  	run_adb shell start b2g
+  	exit $?
+  	;;
+
+  "gaia")
+  	flash_gaia
+  	exit $?
+  	;;
+
+  "time")
+  	update_time
+  	exit $?
+  	;;
+
+  "partition")
+  	sd_partition
+  	exit $?
+  	;;
+
+  "image")
+  	flash_image
```

Comment:
That function makes use of:
 - $2 for the file name which should be passed from $PROJECT
 - $3 for the drive

Here you are passing no arguments, there is no way this can work